### PR TITLE
HS-470: Add IpInterface to Inventory

### PR DESCRIPTION
**Pull request by james8hut**
## Description
Add IpInterface table to Inventory

## Jira link(s)
- https://issues.opennms.org/browse/HS-470

## Flagged for review
Have not renamed MonitoringLocationIntegrationTest to IT as Yang is making some changes in it as per his pull request. Will modify when pull requests settle down for easy merging.

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: inventory/src/main/java/org/opennms/horizon/inventory/mapper/IPInterfaceMapper.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.opennms.horizon.inventory.mapper;
+
+import com.vladmihalcea.hibernate.type.basic.Inet;
+import org.mapstruct.Mapper;
+import org.opennms.horizon.inventory.dto.IpInterfaceDTO;
+import org.opennms.horizon.inventory.model.IpInterface;
+
+@Mapper
```
[the remark this answers is not in the document]
Comment (james8hut) [reply inventory/src/main/java/org/opennms/horizon/inventory/mapper/IPInterfaceMapper.java]: Done

File: inventory/src/main/java/org/opennms/horizon/inventory/rest/IpInterfaceRest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+@RestController
+@RequestMapping("/inventory/ipInterfaces")
+@RequiredArgsConstructor
+public class IpInterfaceRest {
+    private final IpInterfaceService ipInterfaceService;
+
+    @PostMapping
+    public ResponseEntity<IpInterfaceDTO> postIpInterface(@RequestBody IpInterfaceDTO dto) {
+        Optional<IpInterfaceDTO> ml = ipInterfaceService.findIpInterface(dto.getId());
```
Comment (thomasbigger584) [inline inventory/src/main/java/org/opennms/horizon/inventory/rest/IpInterfaceRest.java:27]: what does this variable name `ml` mean? is it ambiguous?

File: inventory/src/main/java/org/opennms/horizon/inventory/rest/IpInterfaceRest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    }
+
+    @GetMapping
+    public ResponseEntity<List<IpInterfaceDTO>> getIpInterface() {
+        List<IpInterfaceDTO> all = ipInterfaceService.findAllIpInterfaces();
+        return new ResponseEntity<>(all, HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/{id}")
```
Comment (thomasbigger584) [inline inventory/src/main/java/org/opennms/horizon/inventory/rest/IpInterfaceRest.java:53]: is `value = ` required here?

File: inventory/src/main/java/org/opennms/horizon/inventory/rest/IpInterfaceRest.java (hunk 2 above)
Comment (james8hut) [reply inventory/src/main/java/org/opennms/horizon/inventory/rest/IpInterfaceRest.java:27]: copy and paste error... will fix

File: inventory/src/main/java/org/opennms/horizon/inventory/rest/IpInterfaceRest.java (hunk 3 above)
Comment (james8hut) [reply inventory/src/main/java/org/opennms/horizon/inventory/rest/IpInterfaceRest.java:53]: possibly not, will fix

**Review by thomasbigger584 — APPROVED** [review]

**Pull request merged**